index.js

```diff
@@ -178,12 +180,11 @@ alexa.request = function(json) {
     }
   };
   this.type = function() {
-    try {
-      return this.data.request.type;
-    } catch (e) {
-      console.error("missing type", e);
-      return null;
+    if (!(this.data && this.data.request && this.data.request.type)) {
+      console.error("missing request type:", this.data);
```

Comment:
Unfortunately I have to put down these alexa PRs and get back to completing some other work for the time being. If someone else wants to pick this up and run with it, should be mostly there. Happy to answer questions. Or I can get back to this next week.